tests/v3_api/test_bkp_restore.py

```diff
@@ -0,0 +1,113 @@
+import base64
+
+import pytest
+from .common import *  # NOQA
+
+CLUSTER_NAME = os.environ.get("RANCHER_CLUSTER_NAME", "")
+namespace = {"p_client": None, "ns": None, "cluster": None, "project": None}
+
+
+def test_bkp_restore():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    client = get_admin_client()
+    cluster = namespace["cluster"]
+    name = random_test_name("default")
+
+    if hasattr(cluster, 'rancherKubernetesEngineConfig'):
+        con = [{"name": "test1",
+                "image": TEST_IMAGE}]
+
+        workload = p_client.create_workload(name=name,
+                                            containers=con,
+                                            namespaceId=ns.id,
+                                            daemonSetConfig={})
+        validate_workload(p_client, workload, "daemonSet", ns.name,
+                          len(get_schedulable_nodes(cluster)))
+        host = "test1.com"
+        path = "/name.html"
+        rule = {"host": host,
+                "paths": [{"workloadIds": [workload.id], "targetPort": "80"}]}
+        p_client.create_ingress(name=name,
+                                namespaceId=ns.id,
+                                rules=[rule])
+        validate_ingress(namespace["p_client"], namespace["cluster"],
+                         [workload], host, path)
+
+        # Perform Backup
+        backup = cluster.backupEtcd()
+        backupname = backup['metadata']['name']
+
+        etcdbackups = cluster.etcdBackups(name=backupname)
+        etcdbackupdata = etcdbackups['data']
+        backupId = etcdbackupdata[0]['id']
+        wait_for_backup_to_active(client, cluster, backupname)
+
+        # Create workload after backup
+        testworkload = p_client.create_workload(name=name,
+                                                containers=con,
+                                                namespaceId=ns.id)
+
+        validate_workload(p_client, testworkload, "deployment", ns.name)
+
+        # Perform Restore
+        cluster.restoreFromEtcdBackup(etcdBackupId=backupId)
+        time.sleep(5)
```

Comment:
Should be able to use validate_cluster() with intermediate_state set to "updating"

Reply:
done